core/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -42,27 +45,43 @@
     private AggregatedDfs aggregatedDfs;
```

Comment:
It's a bit tricky, at least based on my understanding of how things work currently:

1. `ContextIndexSearcher ` is created first inside `SearchService.createContext()`, which is then used to build a `DefaultSearchContext`.  
2. Only after a `DefaultSearchContext` is created can you parse the request and determine if we need to profile or not
3. If profiling is required, we would need to go back and replace the original `ContextIndexSearcher ` with a new `ProfiledIndexSearcher`, but I'm not sure the ramifications of doing that.  It looks like it would have a big cascade effect, since we'd also have to change the searcher inside the created `DefaultSearchContext`

And we can't wrap `ContextIndexSearcher`, because `createWeight()` is called internally by `IndexSearcher` (meaning we'd only be able to profile the top-level of the query).

So the alternative we came up with is this scheme, where we just go back and "tell" the ContextIndexSearcher that it should profile vs not (and it's also why we have all the if/else checks, which isn't ideal either)